src/usenetcreate.py

```
@@ -1073,6 +1073,30 @@ async def prepare_and_upload_usenet(meta: Meta, config: dict[str, Any]) -> str |
+        # overall — it just silently skipped this cleanup and left the temp
+        # files behind. That's the actual bug behind reports of "files
+        # aren't deleted after upload": the case people wouldn't call a
+        # failure is exactly the intermittent-missing-article one.
+        try:
+            if cleanup_upload_root and await aiofiles.ospath.exists(upload_root):
+                if is_debug:
+                    logger.info(f"[cyan][DEBUG SIMULATION] Would delete temporary Usenet folder: {upload_root}[/cyan]")
+                else:
+                    await asyncio.to_thread(shutil.rmtree, upload_root)
+                    logger.info("[green]Cleaned up temporary compressed Usenet files.[/green]")
+        except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)